src/components/calculator/controls/SegmentedControl.tsx

```diff
@@ -14,6 +14,119 @@ import { recordCalculatorOptionSelected } from 'components/Analytics'
 import { CalculatorData } from 'data/calculate'
 import { CheckBoxFormValue } from 'data/data'
 
+export interface SegmentedProps<SourceValue> {
+  id: string
+  header?: string
+  label?: string
+  variant?: string
+  hideRisk?: boolean
+  showDecimals?: boolean
+  verticalOnMobile?: boolean
+  setter: (value: string) => void
+  source: { [key: string]: SourceValue }
+  value: string
+  iconFactory?: (value: string) => JSX.Element
+  multiplierFromSource: (value: SourceValue, key: string) => number
+  labelFromSource: (value: SourceValue, key: string) => string
+  children?: React.ReactNode
+}
+
+// Mobile-friendly segmented button selector (generic version).
+// TODO(jy2wong) the old SegmentedControl still has functionality that this one
+// doesn't; port it over.
+export function SegmentedControlNoDescriptions<SourceValue>(
+  props: SegmentedProps<SourceValue>,
+): React.ReactElement {
+  const { t } = useTranslation()
+  // TODO(jy2wong) formatRiskMultiplier is copy and pasted from
```

Comment:
You should be able to make formatRiskMultiplier a component, which would let you use hooks.

Reply:
done